File: app/models/saga_workflow.py
from datetime import datetime
from mongoengine import Document, StringField, DateTimeField, EnumField
from app.models.enum.SAGAWorkflowStatusEnum import SAGAWorkflowStatusEnum

class SAGA_workflow(Document):
    meta = {'collection': 'saga_workflow'}

    userID = StringField(required=True)
    eventID = StringField(default=None)
    status = EnumField(SAGAWorkflowStatusEnum, default=SAGAWorkflowStatusEnum.PROCESSING)
    created_timestamp = DateTimeField(default=datetime.utcnow)
    ending_timestamp = DateTimeField()

File: eventrio-springboot/eventrio-ai-orchestrator-service/src/main/java/com/eventrio/orchestrator/model/SagaWorkflow.java
package com.eventrio.orchestrator.model;

import com.eventrio.common.enums.SagaWorkflowStatusEnum;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "saga_workflow")
public class SagaWorkflow {

    @Id
    private String id;

    private String userID;

    private String eventID;

    @Builder.Default
    private SagaWorkflowStatusEnum status = SagaWorkflowStatusEnum.PROCESSING;

    @Field("created_timestamp")
    @Builder.Default
    private Instant createdTimestamp = Instant.now();

    @Field("ending_timestamp")
    private Instant endingTimestamp;
}
